rmw_fastrtps_cpp/test/test_logging.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "rmw/rmw.h"
+#include "rmw/error_handling.h"
+
+TEST(TestLogging, rmw_logging)
+{
+  rmw_ret_t ret;
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_DEBUG);
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_INFO);
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_WARN);
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_ERROR);
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_FATAL);
+  EXPECT_EQ(ret, RMW_RET_OK);
```

Comment:
@ahcorde meta: given that we know we're dealing with Fast-RTPS, we can test the side effect using [`GetVerbosity()`](https://www.eprosima.com/docs/fast-rtps/1.6.0/html/classeprosima_1_1fastrtps_1_1_log.html#a32c79f1842be0b3ecd2e9f3a329b2d92).